### DISCUSSION TITLE
Groupings within combobox

### DISCUSSION THREAD

**Discussion by pguptaDS** [twilio-labs/paste#2170] [Office hours] [answered]
Hi!
We are looking to show content by grouping within a combobox. Based on what I am reading, it seems like we can do something like this:
![image](https://user-images.githubusercontent.com/64559618/151853242-fd87f7f9-ee14-4d07-8906-3afffe4dc3a8.png)

but we want to explore something like this:

![image](https://user-images.githubusercontent.com/64559618/151853317-67380ac9-0b43-460c-81f6-c6498b82bcab.png)

The reason is that these lists will become very long as more teams integrate with RBAC and a flat list wouldn't be user frienldy as much. We have a clear need for groupings (by product) so a nested design looks appealing. Nested design also seems very common in other similar tools. How can we achieve that? Thoughts on this?

**Comment by SiTaggart** [comment]
Hi @pguptaDS 

I think this has come up before, and I think it was for the same project to be honest.

Unfortunately, the answer hasn't changed since either https://github.com/twilio-labs/paste/discussions/874

It is not possible to make an accessible, hierarchical Combobox. 

The easiest thing to do to achieve something similar, in a progressive disclosure kind of way, is show dependant Comboboxes. The selection of the first, determines the data in the second. The selection in the second determines the data in the 3rd and so on.

Sorry, I can't give you a better answer, but it's unfortunately just not supported by the specification or browsers.

**Reply by pguptaDS** [reply]
Ah ok. Thanks for the quick response!

**Reply by pguptaDS** [reply]
Would it be possible to have collapsible/expandable rows within combobox? Thats another way to show the products (as top level rows) and the roles underneath if you click on the top row in a progressive manner? @SiTaggart

**Comment by gloriliale** [comment, marked as answer]
Hey @pguptaDS  - just wanted to share the office hours recording with you in case you'd like to revisit any part of the convo. [Here's the video](https://www.loom.com/share/90e25c011a7543108ad2586334475f92), and the password is posted in the #help-design-system channel. Thanks again for coming to office hours!
